cpp/include/cudf/ast/detail/expression_evaluator.cuh

```diff
@@ -29,11 +29,31 @@
 
 #include <rmm/cuda_stream_view.hpp>
 
-namespace cudf {
+#include <cuda/std/type_traits>
+#include <cuda/std/utility>
 
-namespace ast {
+namespace cudf::ast::detail {
 
-namespace detail {
+CUDF_HOST_DEVICE constexpr bool is_complex_type(cudf::type_id type)
```

Comment:
How about nested types? Are they not complex?

Reply:
We don't have benchmarks tracking the performance impact of adding nested types to this complex type group, but based on @matal-nvidia's tests, the presence of nested types does not affect the number of registers used by the kernel. Therefore, we are not including them in this work.